libr/core/linux_heap_glibc.c

```diff
@@ -849,35 +850,38 @@ void GH(print_heap_fastbin)(RCore *core, GHT m_arena, MallocState *main_arena, G
 	}
 }
 
-static void GH(print_tcache_instance)(RCore *core, GHT m_arena, MallocState *main_arena) {
-	if (!core || !core->dbg || !core->dbg->maps) {
-		return;
-	}
-	const int tcache = r_config_get_i (core->config, "dbg.glibc.tcache");
-	if (!tcache) {
-		return;
-	}
-	GHT brk_start = GHT_MAX, brk_end = GHT_MAX, tcache_fd = GHT_MAX, initial_brk = GHT_MAX ;
-	GH(get_brks) (core, &brk_start, &brk_end);
-	GHT tcache_tmp = GHT_MAX, tcache_start = GHT_MAX;
+static void GH(parse_tcache)(RCore *core, GH(RHeapTcache) *tcache_heap) {
+	GHT tcache_fd = GHT_MAX;
+	GHT tcache_tmp = GHT_MAX;
 	RConsPrintablePalette *pal = &r_cons_singleton ()->context->pal;
+	int i;
+	for (i = 0; i < TCACHE_MAX_BINS; i++) {
+		if (tcache_heap->counts[i] > 0) {
+			PRINT_GA("bin :");
+			PRINTF_BA("%2d",i);
+			PRINT_GA(", items :");
+			PRINTF_BA("%2d",tcache_heap->counts[i]);
+			PRINT_GA(", fd :");
 
-	tcache_start = ((brk_start >> 12) << 12) + GH(HDR_SZ);
-	GHT fc_offset = GH(tcache_chunk_size) (core, brk_start);
-	initial_brk = brk_start + fc_offset;
-	if (brk_start == GHT_MAX || brk_end == GHT_MAX || initial_brk == GHT_MAX) {
-		eprintf ("No heap section\n");
-		return;
-	}
-
-	GH(RHeapTcache) *tcache_heap = R_NEW0 (GH(RHeapTcache));
-	if (!tcache_heap) {
-		return;
+			PRINTF_BA("0x%"PFMT64x, (GHT)(tcache_heap->entries[i]) - GH(HDR_SZ));
+			if (tcache_heap->counts[i] > 1) {
+				tcache_fd = (GHT)tcache_heap->entries[i];
+				int n;
+				for(n=1; n < tcache_heap->counts[i]; n++) {
+					(void)r_io_read_at (core->io, tcache_fd, (ut8 *)&tcache_tmp, sizeof (ut64));
+					PRINTF_BA("->0x%"PFMT64x, tcache_tmp - TC_HDR_SZ);
+					tcache_fd = tcache_tmp;
```

Comment:
use the endian apis to read numbers from memory. .this code will only work on machines of the same endian as the target bin